File: 안윤철/깊이, 너비 우선 탐색(DFS, BFS)/타겟넘버.java
class Solution {
    
    private static int answer = 0;
    public int solution(int[] numbers, int target) {
        recursive(numbers, target, 0, 0);
        return answer;
    }
    
    private static void recursive(int[] numbers, int target, int current, int idx) {
        if (numbers.length == idx) {
            if (target == current) {
                answer += 1;
            }
            return;
        }
        
        recursive(numbers, target, current + numbers[idx], idx+1);
        recursive(numbers, target, current - numbers[idx], idx+1);  
    }
}

File: 이예진/깊이, 너비 우선 탐색(DFS, BFS)/타겟넘버.py
def r(numbers, idx, total, target):
    global answer
    if idx == len(numbers):
        if total == target:
            answer += 1
        return
    r(numbers,idx+1, total+numbers[idx],target)
    r(numbers,idx+1, total-numbers[idx],target)
    

def solution(numbers, target):
    global answer
    answer = 0
    r(numbers,0,0,target)
    return answer
